hnn_core/tests/test_network.py

```diff
@@ -443,6 +443,20 @@ def test_network():
             else:
                 assert np.any(np.in1d([item], net.connectivity[conn_idx][arg]))
 
+    # Test searching a list of src or target types
+    cell_type_list = ['L2_basket', 'L5_basket']
+    gid_list = np.array(list(net.gid_ranges['L2_basket']) + list(
+        net.gid_ranges['L5_basket']))
+    indices = pick_connection(net, src_gids=cell_type_list)
+    for conn_idx in indices:
+        assert np.all(np.in1d(
+            net.connectivity[conn_idx]['src_gids'], gid_list))
```

Comment:
Should this test be designed to pass when the two sets of gids are equal (i.e., as opposed to when one is a subset of the other)?

Reply:
I guess the reason I wrote it like this is because the earlier tests check if gids are in a very specific subset can be extracted.

Since we're looping over connections, wouldn't a direct match require pulling the cell type from the connection itself?

Reply:
I think all you'd need to do is append all the `src_gids` into one list, sort them, and compare the final list to `sorted(gid_list)`. If you don't think it's worth it, that's fine - I was just thinking this might help prevent other rare issues with `net.connectivity` and `pick_connection()` by ensuring the output matches 1-to-1 with our expected output.

Reply:
Ah I see! No that's a good idea, I'll see what I can do

Reply:
@rythorpe I decided to match them using `set`(s), let me know if this fits with what you were thinking